File: btjs_page/btjs/mod/checkers/mod/is_range.js


import { is_number } from "./is_number.js"

/*
Function that returns true if the 
data sent is of range array type. 
*/
// return bool
export function is_range(RANGE_ARRAY) {
    //if it is not x 2 array
    if( ! (RANGE_ARRAY.length == 2)) {
        return false
    }
    //if it is not a number
    if( ! is_number(RANGE_ARRAY[0])
    || ! is_number(RANGE_ARRAY[1])) {
        return false
    }
    //if it is not min and max
    if( ! (RANGE_ARRAY[0] < RANGE_ARRAY[1])) {
        return false
    }
    return true
}

File: btpy/src/btpy/btpy_checkers/mod/is_range/is_range.py


def is_range(ANY_DATA:any)-> bool:
    """
    Function that returns true if the 
    data sent is a range in array 
    format, that is, an array of int 
    or float of size x2 where index 0 
    is min and index 1 is max; if the 
    data is not a range, it will return 
    false.
    FINAL
    """
    # Si no es un array de longitud 2
    if (not isinstance(ANY_DATA, list)) \
    or (len(ANY_DATA) != 2):
        return False
    # Si no es un número
    if ((not isinstance(
            ANY_DATA[0], 
            (int, float)
        )) 
    and (isinstance(
            ANY_DATA[1], 
            (int, float)
        ))):
        return False
    # Si no es min y max
    if (not ANY_DATA[0] 
            < ANY_DATA[1]):
        return False
    return True